Editors/Audio/Presentation/AudioEditor/AudioProjectData.cs

```diff
@@ -0,0 +1,141 @@
+        }
+
+        public class DecisionTreeItems
+        {
+            public string StatePath { get; set; }
+            public List<string> AudioFiles { get; set; } = new List<string>();
+        }
+
+        public static void AddAudioProjectToPackFile(PackFileService packFileService, Dictionary<string, List<Dictionary<string, object>>> eventsData)
+        {
+            var audioProjectJson = ConvertEventDataToAudioProject(eventsData);
+            Debug.WriteLine($"audioProjectJson: {audioProjectJson}");
+
+            var pack = packFileService.GetEditablePack();
+            var byteArray = Encoding.ASCII.GetBytes(audioProjectJson);
+            packFileService.AddFileToPack(pack, "AudioProjects", new PackFile($"audio_project.json", new MemorySource(byteArray)));
+        }
+
+        public static string ConvertEventDataToAudioProject(Dictionary<string, List<Dictionary<string, object>>> eventsData)
+        {
+            var audioProject = new JObject();
+
+            var settings = new JObject
+            {
+                ["BnkName"] = "battle_vo_conversational__ovn_vo_actor_Albion_Dural_Durak",
+                ["Language"] = "english(uk)"
+            };
+
+            audioProject["Settings"] = settings;
+
+            var dialogueEvents = new JArray();
+
+            foreach (var eventData in eventsData)
+            {
+                var dialogueEventName = eventData.Key;
+                var eventDataItems = eventData.Value;
+                var decisionTree = new JArray();
+
+                foreach (var eventDataItem in eventDataItems)
+                {
+                    if (eventDataItem.ContainsKey("AudioFiles"))
+                    {
+                        var statePath = eventDataItem
+                            .Where(kv => kv.Key != "AudioFiles" && kv.Key != "AudioFilesDisplay")
+                            .Select(kv => kv.Value.ToString())
+                            .ToList();
+
+                        var decisionBranchItem = new JObject
+                        {
+                            ["StatePath"] = string.Join(".", statePath),
+                            ["AudioFiles"] = new JArray(eventDataItem["AudioFiles"])
+                        };
+
+                        decisionTree.Add(decisionBranchItem);
+                    }
+                }
+
+                var dialogueEvent = new JObject
+                {
+                    ["DialogueEvent"] = dialogueEventName,
+                    ["DecisionTree"] = decisionTree
+                };
+
+                dialogueEvents.Add(dialogueEvent);
+            }
+
+            audioProject["DialogueEvents"] = dialogueEvents;
+
+            return audioProject.ToString(Formatting.Indented);
+        }
+
+        public static Dictionary<string, List<Dictionary<string, object>>> ConvertAudioProjectToEventData(IAudioRepository audioRepository, string audioProjectJson)
```

Comment:
https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/how-to